packages/flutter_tools/lib/src/windows/topaz_ofd_validator.dart

```diff
@@ -14,28 +16,42 @@ class TopazOfdValidator extends DoctorValidator {
 
   @override
   Future<ValidationResult> validate() async {
-    final String tasks = await processLister.getProcessesWithPath('core');
+    final ProcessResult processResult = await processLister.getProcessesWithPath('core');
+    final int exitCode = processResult.exitCode;
+    if (exitCode != 0) {
+      return ValidationResult(
+        ValidationType.missing,
+        <ValidationMessage>[
+          ValidationMessage.hint('Check for Topaz OFD did not complete normally. Returned exit code $exitCode'),
+        ],
+        statusInfo: 'Get-Process failed to complete',
+      );
+    }
+    final String tasks = processResult.stdout as String;
     final RegExp pattern = RegExp(kCoreProcessPattern, multiLine: true, caseSensitive: false);
     final bool matches = pattern.hasMatch(tasks);
     if (matches) {
       return const ValidationResult(
-        ValidationType.partial,
+        ValidationType.missing,
         <ValidationMessage>[
           ValidationMessage.hint('The Topaz OFD Security Module process has been found running. If you are unable to build, you will need to disable it.'),
         ],
         statusInfo: 'Topaz OFD may be running');
     } else {
       return const ValidationResult(
         ValidationType.success,
-        <ValidationMessage>[]);
+        <ValidationMessage>[],
+      );
     }
   }
 }
 
 class ProcessLister {
-  Future<String> getProcessesWithPath(String? filter) async {
+  const ProcessLister();
+
+  Future<ProcessResult> getProcessesWithPath(String? filter) async {
     final String argument = filter != null ? 'Get-Process $filter | Format-List Path' : 'Get-Process | Format-List Path';
-    final ProcessResult taskResult = await Process.run('powershell', <String>['-command', argument]);
-    return taskResult.stdout as String;
+    const ProcessManager processManager = LocalProcessManager();
```

Comment:
let's direct inject this via the constructor. By making this the `LocalProcessManager()` in a local, we could easily accidentally add a "unit" test that ends up spawning real processes on CI.